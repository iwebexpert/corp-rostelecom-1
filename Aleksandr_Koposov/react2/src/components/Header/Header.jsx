import React from 'react'
import { Link } from 'react-router-dom'

import { AppBar, Toolbar, Typography, Button, IconButton, Icon } from '@material-ui/core'

import './Header.scss'

export const Header = (props) => {
  const { children, user } = props
  return (
    <AppBar position="static" className="header">
      <Toolbar>
        <Link className="router__link" to="/">
          <IconButton edge="start" color="inherit">
            <Icon>send</Icon>
          </IconButton>
        </Link>

        <Typography variant="h6" style={{ flexGrow: 1 }}>
          {children}
        </Typography>

        <Link className="router__link" to="/profile">
          <Button
            color="inherit"
            startIcon={<Icon>person</Icon>}
          >
            {(user || {}).name}
          </Button>
        </Link>
      </Toolbar>
    </AppBar>
  )
}


