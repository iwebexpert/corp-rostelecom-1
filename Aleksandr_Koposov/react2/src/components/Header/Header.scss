.header {
    & .router__link {
        color: white;
        text-decoration: none;
    }
}