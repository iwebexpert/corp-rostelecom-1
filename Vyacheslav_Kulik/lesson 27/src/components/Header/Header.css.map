{
"version": 3,
"mappings": "AAAA,OAAQ;EACN,UAAU,EAAE,8GAAqG;EACjH,gBAAgB,EAAE,OAAO;EACzB,OAAO,EAAE,IAAI;EACb,KAAK,EAAE,KAAK",
"sources": ["Header.scss"],
"names": [],
"file": "Header.css"
}
