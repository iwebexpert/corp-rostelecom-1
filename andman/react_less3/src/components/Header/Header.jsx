import React, { Component } from 'react';

export class Header extends Component {

    render() {
        return (
            <div>
                Messenger
            </div>
        );
    }
}