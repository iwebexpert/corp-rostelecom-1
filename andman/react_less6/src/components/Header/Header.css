.menu {
    width:100%;
    display: flex;
    flex-direction: row;    
    justify-content: space-between;
    align-items: center;
    color:white;
}
.ava {
    display: flex;
    flex-direction: row;    
    align-items: center;
}
