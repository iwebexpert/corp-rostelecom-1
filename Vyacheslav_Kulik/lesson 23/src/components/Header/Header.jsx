import React, {Component} from 'react'
import {AppBar, Toolbar, Typography, Grid} from '@material-ui/core'
import './Header.scss'

export class Header extends Component {

    render() {

        return (
                    <Grid container direction="column" alignItems="center" className="header">
                        <Grid item>
                            <Typography >Bot</Typography>
                        </Grid>
                        <Grid item>
                            <Typography>Online</Typography>
                        </Grid>
                    </Grid>
        )

    }

}
