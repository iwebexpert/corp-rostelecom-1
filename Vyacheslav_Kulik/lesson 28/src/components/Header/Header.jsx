import React from 'react'
import {Typography, Grid} from '@material-ui/core'
import {Link} from 'react-router-dom'

import './Header.scss'
//
// export class Header extends Component {
//
//     render() {
//
//         return (
//                     <Grid container direction="column" alignItems="center" className="header">
//                         <Grid item>
//                             <Link to='/profile' style={{textDecoration: 'none', color: 'white'}}>
//                                 <Typography>Your Profile</Typography>
//                             </Link>
//                         </Grid>
//                     </Grid>
//         )
//
//     }
//
// }

export const Header = () => {

    return (
                    <Grid container direction="column" alignItems="center" className="header">
                        <Grid item>
                            <Link to='/profile' style={{textDecoration: 'none', color: 'white'}}>
                                <Typography>Your Profile</Typography>
                            </Link>
                        </Grid>
                    </Grid>
        )
}
