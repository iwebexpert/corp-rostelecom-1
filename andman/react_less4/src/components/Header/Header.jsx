import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './Header.css';

export class Header extends Component {

    render() {
        return (
            <div className="menu">
                Messenger
                <div>
                    <Link to="/profile">Профиль</Link>
                </div>
            </div>
        );
    }
}