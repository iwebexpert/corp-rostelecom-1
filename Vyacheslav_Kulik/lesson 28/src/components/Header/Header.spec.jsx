import React from 'react'
import {Header} from './Header'
import {MemoryRouter} from 'react-router-dom'


describe('Header Test', () => {
    it('should render Header snapshot', function () {
        const container = (<Header/>)
        const wrapper = render(
            <MemoryRouter>
                <Header/>
            </MemoryRouter>
        )
        expect(wrapper).toMatchSnapshot()
    })
})