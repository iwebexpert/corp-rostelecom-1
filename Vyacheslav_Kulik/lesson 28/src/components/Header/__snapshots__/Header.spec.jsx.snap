// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test should render Header snapshot 1`] = `
<div
  class="MuiGrid-root header MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <a
      href="/profile"
      style="text-decoration:none;color:white"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Your Profile
      </p>
    </a>
  </div>
</div>
`;
